python/ray/_raylet.pyx

```diff
@@ -159,6 +159,11 @@ OPTIMIZED = __OPTIMIZE__
 
 logger = logging.getLogger(__name__)
 
+# The currently executing task, if any. These are used to synchronize task
```

Comment:
Also any thoughts about how we can improve the thread model so it's obvious that the code has no deadlock.

Reply:
Probably the best would be if we could figure out a way to enforce that any Python callbacks to CoreWorker don't hold any locks. But I'm not sure how to do that with thread annotations.